upload-chunked-file.js

```diff
@@ -0,0 +1,43 @@
+function chunkFileUpload(file, serverUrl, chunkSize) {
+    const totalChunks = Math.ceil(file.size / chunkSize);
+    let chunkIndex = 0;
+
+    (function uploadNextChunk() {
+        const start = chunkIndex * chunkSize;
+        const end = Math.min(start + chunkSize, file.size);
+        const chunk = file.slice(start, end);
+
+        // Creating a new XMLHttpRequest for every chunk (bad practice)
+        var xhr = new XMLHttpRequest();
+
+        // Opening a new connection every time inside the loop (also not great)
+        xhr.open("POST", serverUrl, true);
+
+        // Setting request header inside the loop for each chunk (redundant)
+        xhr.setRequestHeader("Content-Type", "application/octet-stream");
+        xhr.setRequestHeader("X-Chunk-Index", chunkIndex);
+
+        xhr.onload = function () {
+            if (xhr.status == 200) {
+                console.log("Chunk uploaded:", chunkIndex);
```

Comment:
Refactor the error handling to include retry logic for failed uploads. This can improve the reliability of the file upload process, especially for network-related issues.

```js
function handleError(chunkIndex) {
    // Retry logic with a simple delay
    setTimeout(() => uploadNextChunk(chunkIndex), 1000);
}

xhr.onerror = function () {
    console.error("Network error, retrying chunk upload.");
    handleError(chunkIndex);
};
```